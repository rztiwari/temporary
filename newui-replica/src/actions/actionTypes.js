export const GO_TO_VERIFY = 'GO_TO_VERIFY';
export const LOAD_UI_SCHEMA = 'LOAD_UI_SCHEMA';
export const SUBMIT_PAYMENT_DETAILS = 'SUBMIT_PAYMENT_DETAILS';
export const LOAD_REST_UI_SCHEMA = 'LOAD_REST_UI_SCHEMA';
export const LOAD_ACCOUNTS_LIST = 'LOAD_ACCOUNTS_LIST';
// export const FETCH_ACCOUNTS_LIST = 'FETCH_ACCOUNTS_LIST';
