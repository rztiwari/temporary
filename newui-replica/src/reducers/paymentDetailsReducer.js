import * as types from '../actions/actionTypes';

export default function paymentDetailsReducer(state = {model: "", data: [], resources: {}, accountsList: []}, action) {
  switch(action.type) {
    case types.LOAD_UI_SCHEMA: {
      action.uischema.accountsList = [];
      return action.uischema
    }
    case types.LOAD_REST_UI_SCHEMA: {
      let newState = Object.assign({}, state);
      let newData = action.restUiSchema.data.splice(1);

      if(state.data.length > 2 ){
          newState.data = [state.data[0], state.data[1], ...newData];
      }else {
          newState.data = [...state.data, ...newData ];
      }
      newState.resources = action.restUiSchema.resources;

      if(state.data[0].children == null){
        newState.data[1].childElements.data[2].childElements = action.restUiSchema.data[0].childElements;
      }
      return newState;
    }
    case types.LOAD_ACCOUNTS_LIST: {
      let newState = Object.assign({}, state);
      newState.accountsList = action.accountsListData;
      return newState;
    }
    default:
      return state;
  }
}
