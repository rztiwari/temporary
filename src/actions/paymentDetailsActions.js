import * as types from './actionTypes';
import MMApi from '../middleware/MMApi';

// action creator for going to verify section
export function goToVerify(paymentInstruction) {
  return {type: types.GO_TO_VERIFY, data: paymentInstruction}
}

export function submitPaymentDetails() {
  return {type: types.SUBMIT_PAYMENT_DETAILS}
}

export function loadUiSchema(uischema) {
  return {type: types.LOAD_UI_SCHEMA, uischema}
}

export function loadRestUiSchema(restUiSchema) {
  console.log(restUiSchema);
  return {type: types.LOAD_REST_UI_SCHEMA , restUiSchema}
}

export function fetchRestUiSchema() {
  return (dispatch) => {
    return MMApi.getRestUiSchema()
                .then(uischema => { dispatch(loadRestUiSchema(uischema)) })
                .catch(error => { throw(error) });
  }
}

export function fetchUiSchema() {
  return (dispatch) => {
    return MMApi.getUiSchema()
                .then(uischema => { dispatch(loadUiSchema(uischema.response.view)) })
                .catch(error => { throw(error) });
  }
}

export function loadAccountsList(accountsListData) {
  return {
    type: types.LOAD_ACCOUNTS_LIST,
    accountsListData
  }
}

export function fetchAccountsList() {
  return (dispatch) => {
      return MMApi.getAccountsListData()
                  .then(accountsListData => { return dispatch(loadAccountsList(accountsListData.response.accounts.data)) })
                  .catch(error => {throw(error)});
  }
}
