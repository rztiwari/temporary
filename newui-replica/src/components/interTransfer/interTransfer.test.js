import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';

import InterTransfer from './index';
import RepairNote from '../banners/repairNote';
import * as types from '../../actions/actionTypes';
import * as actions from '../../actions/paymentDetailsActions';
import paymentDetailsReducer from '../../reducers/paymentDetailsReducer';

const middlewares = [ thunk ]
const mockStore = configureMockStore(middlewares)

describe('Intertransfer component', () => {
  it(' renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InterTransfer />, div);
  });

  it('should create an action to load initial data', () => {
    const expectedAction = {
      type: types.LOAD_UI_SCHEMA,
      uischema: []
    }
    expect(actions.loadUiSchema().type).toEqual(expectedAction.type);
  });

  it('creates LOAD_UI_SCHEMA action when fetchUiSchema is dispatched', () => {
    fetchMock.get('*',{
      response: {
        view: []
      }
    });

    const store = mockStore({model: "", data: [], resources: {}, accountsList: []});
    const expectedActions = [{ type: 'LOAD_UI_SCHEMA', uischema: [] }];

    return store.dispatch(actions.fetchUiSchema())
    .then(
      () => {
        expect(store.getActions()).toEqual(expectedActions);
      }
    );
  });

  describe('RepairNote component', () => {
    it('renders correctly', () => {
      const repairnote = renderer.create(<RepairNote />).toJSON();
      expect(repairnote).toMatchSnapshot();
    });
  });

});
