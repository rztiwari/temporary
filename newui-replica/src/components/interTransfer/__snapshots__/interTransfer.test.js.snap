exports[`Intertransfer component RepairNote component renders correctly 1`] = `
<div
  className="repair-note">
  <div
    className="row">
    <div
      className="col-md-1 col-sm-1">
      <i
        className="glyphicon glyphicon-exclamation-sign" />
    </div>
    <div
      className="col-md-11 col-sm-11">
      <h4>
        <b>
          Warning
        </b>
      </h4>
      <ol>
        <li>
          <b>
            HSBC encourages customers to submit payments as early in the day as possible. This will allow for fraud monitoring checks  to be completed and any payment queries to be answered in order to achieve your requested value date.
          </b>
        </li>
        <li>
          <b>
            You may have additional debit accounts that are not listed here. If you cannot find the account you would like to pay from, please return to the payment and transfers tool to create your payment or template. (PBS001)
          </b>
        </li>
      </ol>
    </div>
  </div>
</div>
`;
